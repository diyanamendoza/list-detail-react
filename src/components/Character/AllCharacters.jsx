import { Link } from 'react-router-dom'


export default function AllCharacters({ characters }) {
    return (
        <>
            <ul aria-label="characters">
                {characters.map((character) => (
                <li key={character._id+character.name}>
                    <Link to={`/characters/${character._id}`}>
                    <figure>
                        <img
                        src={character.photoUrl}
                        alt={character.name}
                        />
                        <figcaption>
                        {character.name}
                        </figcaption>
                    </figure>
                    </Link>
                </li>
                ))}
            </ul>
      </>
    )
  }
  