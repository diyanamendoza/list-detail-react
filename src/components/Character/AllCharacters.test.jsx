import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AllCharacters from './AllCharacters';

it('should display a list of characters', () => {
  const { container } = render(
    <MemoryRouter>
      <AllCharacters
        characters={[{
            photoUrl: 'https://placebear.com/200/300',
            name: 'Ted',
            affiliation: 'bear nation',
            allies: 'bees',
            enemies: 'humans',
          },
          {
            photoUrl: 'https://placebear.com/200/300',
            name: 'Fred',
            affiliation: 'bear nation',
            allies: 'bees',
            enemies: 'humans',
          }]}
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
  screen.getByText(/Fred/i)
  screen.getByText(/Ted/i)
});
