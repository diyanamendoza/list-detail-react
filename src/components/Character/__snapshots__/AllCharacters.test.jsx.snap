// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a list of characters 1`] = `
<div>
  <ul
    aria-label="characters"
  >
    <li>
      <a
        href="/characters/undefined"
      >
        <figure>
          <img
            alt="Ted"
            src="https://placebear.com/200/300"
          />
          <figcaption>
            Ted
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/characters/undefined"
      >
        <figure>
          <img
            alt="Fred"
            src="https://placebear.com/200/300"
          />
          <figcaption>
            Fred
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
